src/components/webview/MainWebviewProvider.ts

```diff
@@ -118,42 +120,122 @@ export class IntuitaProvider implements WebviewViewProvider {
 		this.__view.webview.postMessage(message);
 	}
 
-	private __getTreeByCase = (element: Element): TreeNode => {
-		let mappedNode: TreeNode = {
-			id: element.hash,
-			kind: '',
-			children: [],
-		};
+	private __getTreeByCase = (element: Element): TreeNode | undefined => {
+		if (element.kind === ElementKind.ROOT) {
+			element.children.forEach(this.__getTreeByCase);
 
-		mappedNode.label = 'label' in element ? element.label : 'Recipe';
-		mappedNode.iconName = getElementIconBaseName(element.kind);
+			const children: TreeNode[] = [];
 
-		if (element.kind === ElementKind.JOB) {
-			mappedNode = {
-				...mappedNode,
-				...this.__buildJobTree(element),
+			element.children.forEach((caseElement) => {
+				const codemodName = caseElement.codemodName;
+
+				const treeNode = this.__codemodMap.get(codemodName) ?? null;
+				if (treeNode === null) {
+					return;
+				}
+				children.push(treeNode);
+			});
+
+			return {
+				id: element.hash,
+				iconName: getElementIconBaseName(element.kind),
+				label: element.label,
+				kind: 'rootElement',
+				children,
 			};
 		}
 
 		if (element.kind === ElementKind.CASE) {
-			mappedNode = {
-				...mappedNode,
-				...this.__buildCaseTree(element),
-			};
+			if (this.__codemodMap.has(element.codemodName)) {
```

Comment:
this means the elements are up to date?

Reply:
This conditional never should be true. If it is true, that means we have `CASE` elements that have the same codemod name. Will leave this as a comment.